modules/collection/programs/fish.nix

```diff
@@ -0,0 +1,98 @@
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}: let
+  inherit (lib.options) mkOption mkEnableOption mkPackageOption;
+  inherit (lib.strings) typeOf concatMapAttrsStringSep concatMapStringsSep splitString;
+  inherit (lib.modules) mkIf;
+  inherit (lib.types) either str path oneOf attrsOf nullOr bool;
+  inherit (lib.attrsets) mapAttrs' nameValuePair isDerivation;
+
+  cfg = config.rum.programs.fish;
+  env = config.environment.sessionVariables;
+
+  toFish = strOrPath: fileName:
+    if (typeOf strOrPath) == "string"
+    then pkgs.writers.writeFish fileName strOrPath
+    else if isDerivation strOrPath
+    then strOrPath
+    else throw "Input is of invalid type ${typeOf strOrPath}, expected `path` or `string`.";
+
+  toFishFunc = strOrPath: funcName:
+    toFish (
+      if (typeOf strOrPath) == "string"
+      then ''
+        function ${funcName};
+        ${concatMapStringsSep "\n" (line: "\t${line}") (splitString "\n" strOrPath)}
+        end
+      ''
+      else strOrPath
+    ) "${funcName}.fish";
+in {
+  options.rum.programs.fish = {
+    enable = mkEnableOption "fish";
+
+    package = mkPackageOption pkgs "fish" {};
+
+    config = mkOption {
+      default = null;
+      type = nullOr (either str path);
+      description = ''
+        The main configuration for fish, written verbatim to `.config/fish/config.fish`.
+      '';
+    };
+
+    exportEnvVars = mkOption {
+      default = true;
+      type = bool;
+      description = ''
+        Wether to export environment variables set in ''${config.environment.systemVariables}.
+      '';
+    };
+
+    prompt = mkOption {
+      default = null;
+      type = nullOr (either str path);
+      description = ''
+        Your fish prompt, written to `.config/fish/functions/fish_prompt.fish`.
+        It follows the behaviour of `rum.programs.fish.functions`.
+      '';
+    };
```

Comment:
I don't think this is necessary, considering `functions` already exists. What we could do however is add it to the `functions` example, because this is something people might want right off the bat.

Reply:
I shove it into `functions`, so it doesnt add a lot of code and makes prompt, something most (?) people would be using a bit more visible. I'll add it to `functions`' example otherwise.